scripts/eval_attribution.py

```
@@ -0,0 +1,215 @@
+    if not lines[-1].endswith(f"evaluator {expected_evaluator}"):
+        raise RuntimeError(f"Rival did not activate {expected_evaluator}")
+    return int(match.group(1))
+
+
+def searched_score(engine: UciEngine, nodes: int, white_stm: bool) -> int:
+    engine.send("setoption name Clear Hash")
+    engine.send(f"go nodes {nodes}")
+    lines = engine.read_until(lambda line: line.startswith("bestmove "))
+    score_lines = [line for line in lines if re.search(r"\bscore cp -?\d+", line)]
+    if not score_lines:
+        raise RuntimeError("search produced no centipawn score")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)